sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -99,12 +99,35 @@ private object DateTimeFormatterHelper {
     new DateTimeFormatterBuilder().parseCaseInsensitive()
   }
 
-  def toFormatter(builder: DateTimeFormatterBuilder, locale: Locale): DateTimeFormatter = {
-    builder
-      .parseDefaulting(ChronoField.MONTH_OF_YEAR, 1)
-      .parseDefaulting(ChronoField.DAY_OF_MONTH, 1)
-      .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
-      .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
+  def setDefaulting(builder: DateTimeFormatterBuilder,
+      locale: Locale,
+      isWeekBased: Boolean = false): DateTimeFormatterBuilder = {
+
+    val defaults = Seq(
+      (ChronoField.ERA, 1),
+      (ChronoField.MINUTE_OF_HOUR, 0),
+      (ChronoField.SECOND_OF_MINUTE, 0)
+    )
+
+    val weekDayBased = if (isWeekBased) {
+      Seq((ChronoField.DAY_OF_WEEK,
+        WeekFields.of(locale).getFirstDayOfWeek.getValue))
+    } else {
+      Seq(
+        (ChronoField.MONTH_OF_YEAR, 1),
+        (ChronoField.DAY_OF_MONTH, 1))
```

Comment:
I found these default values are very fragile to be set here, so I remove them all in https://github.com/apache/spark/pull/28576 . Can you check if it fixes your problem?